src/test/java/com/google/cloud/tools/appengine/operations/AppEngineWebXmlProjectStagingTest.java

```diff
@@ -75,6 +79,27 @@ public void setUp()
         .run(Mockito.anyList());
   }
 
+  
+  @Test
+  public void testSourceDirectoryRequired() {
+    try {
+      AppEngineWebXmlProjectStageConfiguration.builder().stagingDirectory(destination).build();
+      Assert.fail("allowed missing source directory");
+    } catch (IllegalStateException ex) {
+      Assert.assertEquals("No source directory supplied", ex.getMessage());
+    }
+  }
+  
+  @Test
+  public void testStagingDirectoryRequired() {
+    try {
+      AppEngineWebXmlProjectStageConfiguration.builder().sourceDirectory(destination).build();
```

Comment:
infact the test does it right here?

Reply:
and you noticed it because of the names :-)